rest/rest_collection.cc

```diff
@@ -670,11 +670,12 @@ void
 BackgroundTaskBase::
 cancel()
 {
-    bool wasCancelled = this->cancelled.exchange(true);
```

Comment:
There were a lot of race conditions causing crashes in this code which were fixed with a big amount of effort.  I think we need a stress test to ensure that they stay fixed; this code is very sensitive.